webpack.config.js

```diff
@@ -191,6 +207,11 @@ function getWebpackConfig( { cssFilename, externalizeWordPressPackages = false }
 				name: isDevelopment || shouldEmitStats,
 				maxAsyncRequests: 20,
 				maxInitialRequests: 5,
+				cacheGroups: {
```

Comment:
instead of this one-off for the maps block, maybe we should instead remove the vendors cache group entirely for sdk builds? if / when we use map/mapbox-gl from within calypso, it'll get bundled into the including chunk instead of splitting off to a vendor chunk.

Reply:
For SDK builds, `splitChunks` is already [completely disabled](https://github.com/Automattic/wp-calypso/blob/241ce852572afb8907a3196979e352d621999483/bin/sdk-cli.js#L44). It's the Calypso build (that includes the Gutenpack blocks) that this patch attempts to fix.

Reply:
Yep, this _is_ used in Calypso. https://calypso.live/block-editor/post?branch=fix/clean-public-script
Add a map block (`/map`)
https://calypso.live/block-editor/post?branch=fix/clean-public-script

Reply:
> if / when we use map/mapbox-gl from within calypso, it'll get bundled into the including chunk instead of splitting off to a vendor chunk.

The `mapbox-gl` library is loaded asynchronously (`import().then()`) by the Map block. Because it's huge (600kB). Default webpack `splitChunks` config will prefix the chunk name with `vendors~` because the module is in `node_modules`. This patch just disables the prefix for this one particular chunk. The goal is just to make the name nicer, things would work just fine even without it.

Reply:
Thanks for this clarification, I was a bit confused initially 🙂 

Is there a strong motivation for breaking with the `vendors~` prefix for this chunk? That's really just an indication that the chunk includes no (I'm not sure of the appropriate adjective) "application-specific" code?

Reply:
There's no strong motivation really. It's just the `vendors~map/mapbox-gl` chunk name doesn't make much sense in Calypso. The `map/` folder is there to bundle the Map block stuff together for the Jetpack Blocks build. Ideally, all chunks in Calypso would be in one flat directory. But none of it really matters much.

It was a nice learning experience figuring out how the webpack `cacheGroups` work, but I can revert the resulting webpack.config patch if it causes confusion.

Reply:
I don't have strong feelings, but my preference would be to leave the default vendor prefixing as-is. The webpack config has enough complexity already.

Reply:
OK, I'm removing the fine-tuning of chunk names in 966af9a. Note how that forces me to mention the ugly `vendors~map` name in `.gitignore` and in the `clean:public` NPM script. That was the original (and only) motivation.

Reply:
> Note how that forces me to mention the ugly `vendors~map` name in `.gitignore` and in the `clean:public` NPM script.

This seems to be a symptom of that fact that we don't have any clear division between source and target, but a mix of both living side by side. It would be nice to ignore/clean a directory rather then picking files out of a tree.

Reply:
> This seems to be a symptom of that fact that we don't have any clear division between source and target, but a mix of both living side by side.

Exactly. The `public/` directory is a mix of version-controlled and build files. Once it becomes a build-only directory, we can just clear it and be done.

Reply:
It's tempting to ignore the directory right now and add explicit ignores for those things which are versions. That might work for the `.gitignore` but is probably not a good idea for the clean script. Would look something like this:

```
/public/**
#
# Public contains a mix of versioned files and build results.
# Include versioned files and ignore everything else.
# 
!/public/audio/chat-pling.wav
!/public/fonts/wpeditor.eot
!/public/fonts/wpeditor.svg
!/public/fonts/wpeditor.ttf
!/public/fonts/wpeditor.woff
!/public/images/advanced-seo-nudge.png
!/public/images/authorize/background-1.jpg
!/public/images/authorize/background-2.jpg
!/public/images/authorize/background-3.jpg
!/public/images/authorize/background-4.jpg
!/public/images/authorize/background-5.jpg
!/public/images/authorize/background-6.jpg
!/public/images/checkbox-icons/checkmark.svg
!/public/images/color-schemes/color-scheme-thumbnail-dark.svg
!/public/images/color-schemes/color-scheme-thumbnail-default.svg
!/public/images/color-schemes/color-scheme-thumbnail-light.svg
!/public/images/comments/illustration_comments.svg
!/public/images/comments/illustration_comments_gray.svg
!/public/images/devdocs/calypso-ai-model.png
!/public/images/devdocs/calypso-ai.png
!/public/images/devdocs/calypso-mobile-fit.png
!/public/images/devices/iframe-back.png
!/public/images/drake/drake-all-done.svg
!/public/images/drake/drake-browser.svg
!/public/images/extensions/woocommerce/image-placeholder.png
!/public/images/extensions/woocommerce/paypal.svg
!/public/images/extensions/woocommerce/stripe.svg
!/public/images/extensions/woocommerce/woocommerce-relaxed.svg
!/public/images/extensions/woocommerce/woocommerce-reports.svg
!/public/images/extensions/woocommerce/woocommerce-sample-graph.svg
!/public/images/extensions/woocommerce/woocommerce-setup.svg
!/public/images/extensions/woocommerce/woocommerce-share.svg
!/public/images/extensions/woocommerce/woocommerce-store-creation.svg
!/public/images/favicons/favicon-development.ico
!/public/images/favicons/favicon-horizon.ico
!/public/images/favicons/favicon-staging.ico
!/public/images/favicons/favicon-wpcalypso.ico
!/public/images/g-suite/g-suite.svg
!/public/images/g-suite/logo_docs_48dp.svg
!/public/images/g-suite/logo_drive_48dp.svg
!/public/images/g-suite/logo_gmail_48dp.svg
!/public/images/g-suite/logo_hangouts_48dp.svg
!/public/images/g-suite/logo_hangouts_48px.png
!/public/images/google-my-business/business-local.svg
!/public/images/google-my-business/phone-screenshot-cropped.png
!/public/images/google-my-business/stats-screenshot-cropped.png
!/public/images/google-vouchers/google-voucher.svg
!/public/images/guided-transfer/bluehost-logo.png
!/public/images/guided-transfer/pressable-logo.png
!/public/images/guided-transfer/siteground-logo.png
!/public/images/illustrations/adWords.svg
!/public/images/illustrations/ads-removed.svg
!/public/images/illustrations/advertisingRemoved.svg
!/public/images/illustrations/al-syncing-site.svg
!/public/images/illustrations/almost-there.svg
!/public/images/illustrations/anti-spam.svg
!/public/images/illustrations/app-banner-editor.svg
!/public/images/illustrations/app-banner-notifications.svg
!/public/images/illustrations/app-banner-reader.svg
!/public/images/illustrations/app-banner-stats.svg
!/public/images/illustrations/backup.svg
!/public/images/illustrations/business-address.svg
!/public/images/illustrations/checkEmailsDesktop.svg
!/public/images/illustrations/checkEmailsPhone.svg
!/public/images/illustrations/clone-site-origin.svg
!/public/images/illustrations/cloudBased.svg
!/public/images/illustrations/contact-us.svg
!/public/images/illustrations/custom-domain-blogger.svg
!/public/images/illustrations/custom-domain.svg
!/public/images/illustrations/customizeTheme-onDark.svg
!/public/images/illustrations/customizeTheme.svg
!/public/images/illustrations/error.svg
!/public/images/illustrations/fireworks.svg
!/public/images/illustrations/gauge.svg
!/public/images/illustrations/get-apps-banner.svg
!/public/images/illustrations/google-adwords.svg
!/public/images/illustrations/google-my-business-feature.svg
!/public/images/illustrations/gutenberg-mini.svg
!/public/images/illustrations/gutenberg.svg
!/public/images/illustrations/happiness-support.svg
!/public/images/illustrations/homepage-news.svg
!/public/images/illustrations/homepage-static.svg
!/public/images/illustrations/illustration-404.svg
!/public/images/illustrations/illustration-conversations-empty.svg
!/public/images/illustrations/illustration-empty-results.svg
!/public/images/illustrations/illustration-jetpack.svg
!/public/images/illustrations/illustration-layout.svg
!/public/images/illustrations/illustration-nosites.svg
!/public/images/illustrations/illustration-ok.svg
!/public/images/illustrations/illustration-shopping-bags.svg
!/public/images/illustrations/illustration-start.svg
!/public/images/illustrations/illustration-write.svg
!/public/images/illustrations/install-button.svg
!/public/images/illustrations/jetpack-apps.svg
!/public/images/illustrations/jetpack-business-address.svg
!/public/images/illustrations/jetpack-concierge.svg
!/public/images/illustrations/jetpack-connection-migration.svg
!/public/images/illustrations/jetpack-connection.svg
!/public/images/illustrations/jetpack-dashboard.svg
!/public/images/illustrations/jetpack-google-analytics.svg
!/public/images/illustrations/jetpack-header.svg
!/public/images/illustrations/jetpack-marketing.svg
!/public/images/illustrations/jetpack-payments.svg
!/public/images/illustrations/jetpack-search.svg
!/public/images/illustrations/jetpack-security.svg
!/public/images/illustrations/jetpack-site-activity.svg
!/public/images/illustrations/jetpack-spam.svg
!/public/images/illustrations/jetpack-speed.svg
!/public/images/illustrations/jetpack-start.svg
!/public/images/illustrations/jetpack-stats.svg
!/public/images/illustrations/jetpack-support.svg
!/public/images/illustrations/jetpack-themes.svg
!/public/images/illustrations/jetpack-updates.svg
!/public/images/illustrations/jetpack-video-hosting.svg
!/public/images/illustrations/jetpack-wordads.svg
!/public/images/illustrations/migrating-host-assisted.svg
!/public/images/illustrations/migrating-host-diy.svg
!/public/images/illustrations/no-themes-drake.svg
!/public/images/illustrations/plan-business-circle.svg
!/public/images/illustrations/plan-business.svg
!/public/images/illustrations/plan-free-circle.svg
!/public/images/illustrations/plan-free.svg
!/public/images/illustrations/plan-personal-circle.svg
!/public/images/illustrations/plan-personal.svg
!/public/images/illustrations/plan-premium-circle.svg
!/public/images/illustrations/plan-premium.svg
!/public/images/illustrations/plugins-big.svg
!/public/images/illustrations/plugins-small.svg
!/public/images/illustrations/rebrand-cities-welcome.svg
!/public/images/illustrations/security-issue.svg
!/public/images/illustrations/security.svg
!/public/images/illustrations/site-title.svg
!/public/images/illustrations/stats.svg
!/public/images/illustrations/support.svg
!/public/images/illustrations/thankYou.svg
!/public/images/illustrations/type-blog.svg
!/public/images/illustrations/type-business.svg
!/public/images/illustrations/type-e-commerce.svg
!/public/images/illustrations/type-personal.svg
!/public/images/illustrations/type-portfolio.svg
!/public/images/illustrations/type-website.svg
!/public/images/illustrations/upsell-enable.svg
!/public/images/illustrations/upsell-payments.svg
!/public/images/illustrations/upsell-show-ads.svg
!/public/images/illustrations/video-hosting-13G.svg
!/public/images/illustrations/video-hosting-unlimited.svg
!/public/images/illustrations/videoAudioPosts.svg
!/public/images/illustrations/waitTime.svg
!/public/images/illustrations/whoops.svg
!/public/images/illustrations/wordAds.svg
!/public/images/jetpack/illustration-jetpack-monitor.svg
!/public/images/jetpack/illustration-jetpack-protect.svg
!/public/images/jetpack/illustration-jetpack-scan.svg
!/public/images/jetpack/install-wp-admin.svg
!/public/images/jetpack/jetpack-bluehost-connection.png
!/public/images/jetpack/jetpack-dreamhost-connection.png
!/public/images/jetpack/jetpack-logo.svg
!/public/images/jetpack/jetpack-manage.svg
!/public/images/jetpack/jetpack-milesweb-connection.png
!/public/images/jetpack/jetpack-pressable-connection.png
!/public/images/jetpack/powered-by-jetpack.svg
!/public/images/login/check-email.svg
!/public/images/login/pushauth.svg
!/public/images/manifest/icon-144x144.png
!/public/images/manifest/icon-192x192.png
!/public/images/manifest/icon-512x512.png
!/public/images/me/get-apps-google-play.png
!/public/images/me/get-apps-ios-store.svg
!/public/images/me/pattern-dark.png
!/public/images/media/illustration-media.svg
!/public/images/pages/illustration-pages.svg
!/public/images/pages/photolia.jpg
!/public/images/people/mystery-person.svg
!/public/images/plans/jetpack/plan-beginner.svg
!/public/images/plans/jetpack/plan-business-circle.svg
!/public/images/plans/jetpack/plan-business.svg
!/public/images/plans/jetpack/plan-free-circle.svg
!/public/images/plans/jetpack/plan-personal-circle.svg
!/public/images/plans/jetpack/plan-personal.svg
!/public/images/plans/jetpack/plan-premium-circle.svg
!/public/images/plans/jetpack/plan-premium.svg
!/public/images/plans/wpcom/plan-beginner.svg
!/public/images/plans/wpcom/plan-blogger-circle.svg
!/public/images/plans/wpcom/plan-blogger.svg
!/public/images/plans/wpcom/plan-business-circle.svg
!/public/images/plans/wpcom/plan-business.svg
!/public/images/plans/wpcom/plan-ecommerce-circle.svg
!/public/images/plans/wpcom/plan-ecommerce.svg
!/public/images/plans/wpcom/plan-free-circle.svg
!/public/images/plans/wpcom/plan-free.svg
!/public/images/plans/wpcom/plan-personal-circle.svg
!/public/images/plans/wpcom/plan-personal.svg
!/public/images/plans/wpcom/plan-premium-circle.svg
!/public/images/plans/wpcom/plan-premium.svg
!/public/images/plugins/plugins_hero.svg
!/public/images/posts/illustration-posts.svg
!/public/images/private/private.svg
!/public/images/reader/play-icon.png
!/public/images/reader/promo-app-icon.png
!/public/images/reader/reader-conversations-characters.svg
!/public/images/reader/reader-conversations.png
!/public/images/reader/reader-discover.png
!/public/images/reader/reader-intro-background.svg
!/public/images/reader/reader-intro-character.svg
!/public/images/reader/reader-mobile.png
!/public/images/reader/reader-notifications.png
!/public/images/reader/reader-welcome-illustration.svg
!/public/images/rebrand-cities/wp-rebrand-cities-logo.svg
!/public/images/related-posts/cat-blog.png
!/public/images/related-posts/devices.jpg
!/public/images/related-posts/mobile-wedding.jpg
!/public/images/seo/page-title.svg
!/public/images/sharing/eventbrite-list.png
!/public/images/sharing/eventbrite-widget.png
!/public/images/sharing/facebook-profile.png
!/public/images/sharing/facebook-sharing.png
!/public/images/sharing/google-photos.png
!/public/images/sharing/google-publicize.png
!/public/images/sharing/google-sharing.png
!/public/images/sharing/instagram-widget.png
!/public/images/sharing/linkedin-publicize.png
!/public/images/sharing/linkedin-sharing.png
!/public/images/sharing/mailchimp-screenshot.png
!/public/images/sharing/tumblr-publicize.png
!/public/images/sharing/tumblr-sharing.png
!/public/images/sharing/twitter-publicize.png
!/public/images/sharing/twitter-timeline.png
!/public/images/signup/confetti.svg
!/public/images/signup/seo-guy.svg
!/public/images/simple-payments/paypal-logo.svg
!/public/images/stats/chart-today.png
!/public/images/stats/chart.png
!/public/images/stats/chart2.png
!/public/images/stats/date-picker.png
!/public/images/stats/illustration-stats-intro.svg
!/public/images/stats/illustration-stats.svg
!/public/images/stats/left-arrow.svg
!/public/images/stats/map.jpg
!/public/images/stats/right-arrow.svg
!/public/images/stats/search-engine.png
!/public/images/stats/stats-geo-chart.png
!/public/images/stats/tasks/about.svg
!/public/images/stats/tasks/contact.svg
!/public/images/stats/tasks/create-tagline.svg
!/public/images/stats/tasks/custom-domain.svg
!/public/images/stats/tasks/domains.svg
!/public/images/stats/tasks/first-post.svg
!/public/images/stats/tasks/launch.svg
!/public/images/stats/tasks/mobile-app.svg
!/public/images/stats/tasks/personalize-your-site.svg
!/public/images/stats/tasks/ready-to-share.svg
!/public/images/stats/tasks/social-links.svg
!/public/images/stats/tasks/upload-icon.svg
!/public/images/stats/tasks/upload-profile-picture.svg
!/public/images/themes-banner/photo-blog.png
!/public/images/themes-banner/small-business.png
!/public/images/tinymce-icons/gridicons-chevron-down.svg
!/public/images/upgrades/advertising-removed.svg
!/public/images/upgrades/adwords.svg
!/public/images/upgrades/alipay.svg
!/public/images/upgrades/apple.svg
!/public/images/upgrades/bancontact.svg
!/public/images/upgrades/brazil-tef.svg
!/public/images/upgrades/cc-amex-disabled.svg
!/public/images/upgrades/cc-amex.svg
!/public/images/upgrades/cc-cvv-back.svg
!/public/images/upgrades/cc-cvv-front.svg
!/public/images/upgrades/cc-diners-disabled.svg
!/public/images/upgrades/cc-diners.svg
!/public/images/upgrades/cc-discover-disabled.svg
!/public/images/upgrades/cc-discover.svg
!/public/images/upgrades/cc-jcb-disabled.svg
!/public/images/upgrades/cc-jcb.svg
!/public/images/upgrades/cc-mastercard-disabled.svg
!/public/images/upgrades/cc-mastercard.svg
!/public/images/upgrades/cc-placeholder.svg
!/public/images/upgrades/cc-unionpay-disabled.svg
!/public/images/upgrades/cc-unionpay.svg
!/public/images/upgrades/cc-visa-disabled.svg
!/public/images/upgrades/cc-visa.svg
!/public/images/upgrades/check-emails-desktop.svg
!/public/images/upgrades/custom-domain.svg
!/public/images/upgrades/customize-theme.svg
!/public/images/upgrades/emergent-paywall.svg
!/public/images/upgrades/eps.svg
!/public/images/upgrades/g-suite-logo.png
!/public/images/upgrades/giropay.svg
!/public/images/upgrades/ideal.svg
!/public/images/upgrades/items-failed.svg
!/public/images/upgrades/media-post.svg
!/public/images/upgrades/p24.svg
!/public/images/upgrades/paypal-disabled.svg
!/public/images/upgrades/paypal.svg
!/public/images/upgrades/paytm.svg
!/public/images/upgrades/plugins.svg
!/public/images/upgrades/publish-button.svg
!/public/images/upgrades/tef.svg
!/public/images/upgrades/thank-you.svg
!/public/images/upgrades/wait-time.svg
!/public/images/upgrades/wechat.svg
!/public/images/upgrades/word-ads.svg
!/public/images/woo/ninja-joy.svg
!/public/images/wordpress/logo-jpc.svg
!/public/images/wordpress/logo-stars.svg
!/public/tinymce/skins/lightgray/content.inline.min.css
!/public/tinymce/skins/lightgray/content.min.css
!/public/tinymce/skins/lightgray/fonts/tinymce-small.eot
!/public/tinymce/skins/lightgray/fonts/tinymce-small.svg
!/public/tinymce/skins/lightgray/fonts/tinymce-small.ttf
!/public/tinymce/skins/lightgray/fonts/tinymce-small.woff
!/public/tinymce/skins/lightgray/fonts/tinymce.eot
!/public/tinymce/skins/lightgray/fonts/tinymce.svg
!/public/tinymce/skins/lightgray/fonts/tinymce.ttf
!/public/tinymce/skins/lightgray/fonts/tinymce.woff
!/public/tinymce/skins/lightgray/img/anchor.gif
!/public/tinymce/skins/lightgray/img/loader.gif
!/public/tinymce/skins/lightgray/img/object.gif
!/public/tinymce/skins/lightgray/img/trans.gif
!/public/tinymce/skins/lightgray/skin.min.css
!/public/tinymce/skins/woocommerce/content.css
!/public/tinymce/skins/woocommerce/editor.css
!/public/tinymce/skins/wordpress/images/audio.png
!/public/tinymce/skins/wordpress/images/dashicon-edit.png
!/public/tinymce/skins/wordpress/images/dashicon-no.png
!/public/tinymce/skins/wordpress/images/embedded.png
!/public/tinymce/skins/wordpress/images/gallery-2x.png
!/public/tinymce/skins/wordpress/images/gallery.png
!/public/tinymce/skins/wordpress/images/more-2x.png
!/public/tinymce/skins/wordpress/images/more.png
!/public/tinymce/skins/wordpress/images/pagebreak-2x.png
!/public/tinymce/skins/wordpress/images/pagebreak.png
!/public/tinymce/skins/wordpress/images/playlist-audio.png
!/public/tinymce/skins/wordpress/images/playlist-video.png
!/public/tinymce/skins/wordpress/images/video.png
!/public/tinymce/skins/wordpress/wp-content.css
```